mitodl/ocw-studio
# Collaboration UI

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #60 

#### What's this PR do?
Adds a UI for listing and modifying website collaborators

#### How should this be manually tested?
- Log in as a superuser or website admin or owner and go to `http://sites/<website_name>`
- Click on the collaborators link.  You should see a list of collaborators (if any) and an 'Add collaborators' button.  Any collaborators who are not global admins or the website owner should have edit and delete icons next to them.
- Click the 'Add collaborator' button.  Fill out the form, try out validation by submitting with bad/missing data, then enter valid input and click Submit.  You should end up back on the collaborators list with the new collaborator visible.
- Click the edit icon for the new collaborator.  Change the role to --- and submit, you should get a validation error.  Change the role to 'Administrator' or 'Editor' and submit.  You should end up back on the list, with the change reflected in the list.
- Click the delete icon for the new collaborator.  You should get a confirmation dialog.  Click 'Delete'.  The collaborator should no longer appear on the list.
- Log in as an editor for the site (you can create one first if necessary).  Go to `http://sites/<website_name>` - there should not be a 'Collaborators' link.


#### Screenshots (if appropriate)
<img width="948" alt="Screen Shot 2021-02-26 at 3 45 47 PM" src="https://user-images.githubusercontent.com/187676/109352982-b9cb8280-7849-11eb-9c2c-fef189e5d2f8.png">


<img width="1069" alt="Screen Shot 2021-02-25 at 12 42 25 PM" src="https://user-images.githubusercontent.com/187676/109195624-bd3b0d00-7768-11eb-8994-f6e1ef9821f6.png">

<img width="1070" alt="Screen Shot 2021-02-25 at 12 42 05 PM" src="https://user-images.githubusercontent.com/187676/109195636-c0ce9400-7768-11eb-879c-da05fca86023.png">

<img width="1070" alt="Screen Shot 2021-02-25 at 12 41 54 PM" src="https://user-images.githubusercontent.com/187676/109195643-c330ee00-7768-11eb-9548-ed23b37f9a23.png">


#### What GIF best describes this PR or how it makes you feel?
(Optional)
